### PR TITLE
[RFR] Allow to customize fields

### PR DESCRIPTION
**Pull request by manuquentin**
- [x] Change `Compile` component to accept `props` and `functions`
- [x] Create a `FieldViewConfiguration` and expose it to configuration
- [x] Use `Column` and `Field` in List/Create/Edit views
- [x] Add tests

### CODE REVIEW

File: app/Field/BooleanFieldView.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class BooleanFieldView {
+    static getReadWidget() {
+        return '<BooleanColumn value={this.props.value} />';
```
Comment (fzaninotto) [inline app/Field/BooleanFieldView.js]: I don't understand why you return a string and not a JSX. You could avoid compile altogether by using JSX. And it's more React-y.

Comment (fzaninotto) [inline app/Field/BooleanFieldView.js]: Bump. Is it possible to return components, no strings?

Comment (manuquentin) [inline app/Field/BooleanFieldView.js]: I'm on it, found a way but I must dig a little more

File: app/Component/Column/StringColumn.js [hunk 2]
```diff
@@ -2,19 +2,14 @@ import React from 'react';
 
 class StringColumn extends React.Component {
     render() {
-        let {value, detailAction} = this.props;
```
Comment (manuquentin) [inline app/Component/Column/StringColumn.js]: Link is handled by the `Column` component

File: app/Component/Field/Field.js [hunk 3]
```diff
@@ -1,37 +1,29 @@
...
-            case 'boolean':
-                renderedField = <CheckboxField name={fieldName} value={value} updateField={this.props.updateField} />;
-                break;
-
-            default:
-                // TODO: add more types
-                renderedField = ''; // temporay
-                //throw new Error(`Unknown field type "${field.type()}".`);
-        }
+        let fieldTemplate = fieldView ? fieldView.getWriteWidget : null;
```
Comment (manuquentin) [inline app/Component/Field/Field.js]: We pass a function directly to the `Compile` directive because it should be called at the compile time

Comment (jeromemacias) [inline app/Component/Field/Field.js]: Should'nt you call `fieldView.getWriteWidget()` ?

Comment (jeromemacias) [inline app/Component/Field/Field.js]: Ok, You will call the function into the `Compile` component ?

File: app/Component/__tests__/DashboardPanelTest.js [hunk 4]
```diff
@@ -57,28 +76,28 @@ describe('DashboardPanel', () => {
...
+        //it('should set header with clickable label', () => {
+        //    var panel = getPanel(view, entity.label(), dataStore, null, null);
+        //    panel = React.findDOMNode(panel);
+        //
+        //    var list = panel.getElementsByClassName('panel-heading')[0].querySelector('a');
+        //    TestUtils.Simulate.click(list);
+        //
+        //    expect(list.attributes['data-click-to'].value).toEqual('list');
+        //    expect(list.attributes['data-params'].value).toEqual('{"entity":"posts"}');
+        //});
```
Comment (jeromemacias) [inline app/Component/__tests__/DashboardPanelTest.js]: Why this part was commented ?

File: app/Component/__tests__/DashboardPanelTest.js [hunk 5]
```diff
@@ -57,28 +76,28 @@ describe('DashboardPanel', () => {
...
+    //        var panel = getPanel(view, entity.label(), dataStore, null, null);
+    //        panel = React.findDOMNode(panel);
+    //
+    //        var rows = panel.querySelectorAll('table tbody tr');
+    //
+    //        expect(rows.length).toEqual(1);
+    //        expect(rows[0].childNodes.length).toEqual(3);
+    //        expect(rows[0].childNodes[1].textContent).toEqual('First Post');
+    //    });
+    //});
```
Comment (jeromemacias) [inline app/Component/__tests__/DashboardPanelTest.js]: Same here.

File: app/Component/Field/Field.js (hunk 3 above)
Comment (manuquentin) [inline app/Component/Field/Field.js]: Yep

File: app/Component/__tests__/DashboardPanelTest.js (hunk 5 above)
Comment (manuquentin) [inline app/Component/__tests__/DashboardPanelTest.js]: A mistake, please refresh

**Pull request merged**